packages/lean_analytics/lib/core/route_observer.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:lean_analytics/core/core.dart';
+import 'package:provider/provider.dart';
+
+abstract class LeanRoute extends MaterialPageRoute<void> {
+  LeanRoute(
+    this.name, {
+    required super.builder,
+    required super.settings,
+  });
+
+  final String name;
+}
+
+class GoRouterNavigationObserver extends NavigatorObserver {
+  GoRouterNavigationObserver(this.context);
+
+  final BuildContext context;
```

Comment:
Instead of persisting build context (which can go stale), wouldnt it be better to just inject LeanAnalytics?